### PR TITLE
HOTFIX: Fix tag non hashable

### LINKED ISSUE #1131: Non hashable Tags on SDF file will cause ligand creation from and  a `Ligand`
SDF tags that look like a list or other unhashable type will cause the following to fail. 

```python
        tags = {(key, value) for key, value in kwargs.items() if key in keys_to_save}
```

Lets just skip them for now with a warning, to allow project support to move forward.

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #1131 

## Developers certificate of origin
- [X] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by hmacdope** [comment]
@apayne97 @kaminow could I grab a review here?

File: asapdiscovery-data/asapdiscovery/data/schema/ligand.py [hunk 1]
```diff
@@ -215,13 +215,30 @@ def from_oemol(cls, mol: oechem.OEMol, **kwargs) -> "Ligand":
...
+                        f"Tag {key} with value {value} is not hashable and will not be saved"
+                    )
+
         kwargs["tags"] = tags
 
         # Do the same thing for the conformer tags, only keeping the ones in 'tags'
-        kwargs["conf_tags"] = [
-            (key, value) for key, value in conf_tags.items() if key in keys_to_save
-        ]
+        conf_tags_lst = []
```
Comment (kaminow) [inline asapdiscovery-data/asapdiscovery/data/schema/ligand.py:233]:
```suggestion
        conf_tags_list = []
```

File: asapdiscovery-data/asapdiscovery/data/schema/ligand.py [hunk 2]
```diff
@@ -215,13 +215,30 @@ def from_oemol(cls, mol: oechem.OEMol, **kwargs) -> "Ligand":
...
+        for key, value in conf_tags.items():
+            if key in keys_to_save:
+                try:
+                    conf_tags_lst.append((key, value))
+                except TypeError:
+                    pass
```
Comment (kaminow) [inline asapdiscovery-data/asapdiscovery/data/schema/ligand.py:239]: add a warning here too?

File: asapdiscovery-data/asapdiscovery/data/schema/ligand.py [hunk 3]
```diff
@@ -215,13 +215,30 @@ def from_oemol(cls, mol: oechem.OEMol, **kwargs) -> "Ligand":
         keys_to_save = [
             key for key in kwargs.keys() if key not in cls.__fields__.keys()
         ]
-        tags = {(key, value) for key, value in kwargs.items() if key in keys_to_save}
+
+        tags = set()
```
Comment (kaminow) [inline asapdiscovery-data/asapdiscovery/data/schema/ligand.py:219]: if you want to I think you could put this back as a list comprehension with `isinstance(..., collections.abc.Hashable)` but this is fine too, up to you

File: asapdiscovery-data/asapdiscovery/data/schema/ligand.py [hunk 4]
```diff
@@ -215,13 +215,30 @@ def from_oemol(cls, mol: oechem.OEMol, **kwargs) -> "Ligand":
...
-        ]
+        conf_tags_lst = []
+        for key, value in conf_tags.items():
+            if key in keys_to_save:
+                try:
+                    conf_tags_lst.append((key, value))
+                except TypeError:
+                    pass
+
+        kwargs["conf_tags"] = conf_tags_lst
```
Comment (kaminow) [inline asapdiscovery-data/asapdiscovery/data/schema/ligand.py:241]:
```suggestion
        kwargs["conf_tags"] = conf_tags_list
```

File: asapdiscovery-data/asapdiscovery/data/schema/ligand.py [hunk 5]
```diff
@@ -215,13 +215,30 @@ def from_oemol(cls, mol: oechem.OEMol, **kwargs) -> "Ligand":
...
 
         # Do the same thing for the conformer tags, only keeping the ones in 'tags'
-        kwargs["conf_tags"] = [
-            (key, value) for key, value in conf_tags.items() if key in keys_to_save
-        ]
+        conf_tags_lst = []
+        for key, value in conf_tags.items():
+            if key in keys_to_save:
+                try:
+                    conf_tags_lst.append((key, value))
```
Comment (kaminow) [inline asapdiscovery-data/asapdiscovery/data/schema/ligand.py:237]:
```suggestion
                    conf_tags_list.append((key, value))
```

Comment (kaminow) [inline asapdiscovery-data/asapdiscovery/data/schema/ligand.py:237]: if this is just a list do we still need the `try`?

**Review by kaminow — APPROVED** [review]
looks good, just some minor stuff but feel free to merge when you're happy

**Pull request merged**